VirtIO/VirtIOPCILegacy.c

```diff
@@ -160,7 +160,7 @@ static NTSTATUS vio_legacy_query_vq_alloc(VirtIODevice *vdev,
 
     /* Check if queue is either not available or already active. */
     num = ioread16(vdev, vdev->addr + VIRTIO_PCI_QUEUE_NUM);
-    if (!num || ioread32(vdev, vdev->addr + VIRTIO_PCI_QUEUE_PFN)) {
+    if (!num || !ioread32(vdev, vdev->addr + VIRTIO_PCI_QUEUE_PFN)) {
```

Comment:
Can you please describe how you hit the problem that requires the change?
Current code checks how many queues are active and if for some reason the queue address is already set.

Reply:
Also if the change is needed, also the comment for the code should change

Reply:
> Can you please describe how you hit the problem that requires the change? Current code checks how many queues are active and if for some reason the queue address is already set.

I allocate a leagacy vsock device to windows, and windows start, then I install the vsock driver of virtio-win, the driver will has error and fail to install. The error is This device cannot start (Code 10) STATUS_DEVICE_POWER_FAILURE.
Then I use logman and wpa to record and analyze the trace, found the error trace: VIOSockRxVqInit virtio_query_queue_allocation(VIOSOCK_VQ_RX) failed. Then I go into the function virtio_query_queue_allocation->vio_legacy_query_vq_alloc, then find ioread32(vdev, vdev->addr + VIRTIO_PCI_QUEUE_PFN) return not 0 thats cause vio_legacy_query_vq_alloc return STATUS_NOT_FOUND, so it will affect the vsock driver install.